ssc/src/Pos/Ssc/Worker.hs

```diff
@@ -232,7 +241,7 @@ onNewSlotOpening pm params SlotId {..} sendOpening
         mbOpen' <- case params of
             SscOpeningNone   -> pure Nothing
             SscOpeningNormal -> pure (Just open)
-            SscOpeningWrong  -> Just <$> liftIO (QC.generate QC.arbitrary)
+            SscOpeningWrong  -> Just <$> liftIO randomOpening
```

Comment:
This is a lot better, certainly. But.. *why* are we generating a random opening here? This seems deeply weird to me.

Reply:
I'm going to be digging into this today to really understand.